sdk/core/Azure.Core.Experimental/src/ContentType.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Represents content type.
+    /// </summary>
+    public readonly struct ContentType : IEquatable<ContentType>, IEquatable<string>
```

Comment:
@pakrym / @chamons - could code gen make use of this type for produces/consumes instead of generating your own?

Reply:
Yes